### PR TITLE
z_bgcheck OK

### PR DESCRIPTION
**Pull request by Synray**

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]
The migrated bss could be a bit problematic in the future because some of it is in-function-static.
Hopefully it won't break too much

File: src/code/z_bgcheck.c [hunk 1]
```diff
@@ -1947,13 +1912,14 @@ s32 BgCheck_CheckWallImpl(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResul
     dz = posNext->z - posPrev->z;
 
     // if there's movement on the xz plane, and argA flag is 0,
-    if (((dx != 0.0f) || (dz != 0.0f)) && !(argA & 1)) {
-        if ((checkHeight + dy) < 5.0f) {
+    if ((dx != 0.0f || dz != 0.0f) && !(argA & 1)) {
+        if (checkHeight + dy < 5.0f) {
```
Comment (engineer124) [inline src/code/z_bgcheck.c:1916]:
```suggestion
    if (((dx != 0.0f) || (dz != 0.0f)) && !(argA & 1)) {
        if ((checkHeight + dy) < 5.0f) {
```
These brackets were dropped but we're trying to preserve brackets

File: src/code/z_bgcheck.c [hunk 2]
```diff
@@ -1947,13 +1912,14 @@ s32 BgCheck_CheckWallImpl(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResul
...
             //! @bug checkHeight is not applied to posPrev/posNext
             result = BgCheck_CheckLineImpl(colCtx, xpFlags, COLPOLY_IGNORE_NONE, posPrev, posNext, &posIntersect, &poly,
                                            &bgId, actor, 1.0f, BGCHECK_CHECK_ALL & ~BGCHECK_CHECK_CEILING);
             if (result) {
-                ny = COLPOLY_GET_NORMAL(poly->normal.y);
+                // unit normal of polygon
+                f32 ny = COLPOLY_GET_NORMAL(poly->normal.y);
                 // if poly is floor, push result underneath the floor
```
Comment (engineer124) [inline src/code/z_bgcheck.c:1923]:
```suggestion
                f32 ny = COLPOLY_GET_NORMAL(poly->normal.y);

                // if poly is floor, push result underneath the floor
```
Spaces after declarations

File: src/code/z_bgcheck.c [hunk 3]
```diff
@@ -1969,8 +1935,8 @@ s32 BgCheck_CheckWallImpl(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResul
                 }
                 // poly is wall
                 else {
-                    nx = COLPOLY_GET_NORMAL(poly->normal.x);
-                    nz = COLPOLY_GET_NORMAL(poly->normal.z);
+                    f32 nx = COLPOLY_GET_NORMAL(poly->normal.x);
+                    f32 nz = COLPOLY_GET_NORMAL(poly->normal.z);
                     posResult->x = radius * nx + posIntersect.x;
```
Comment (engineer124) [inline src/code/z_bgcheck.c:1940]:
```suggestion
                    f32 nz = COLPOLY_GET_NORMAL(poly->normal.z);

                    posResult->x = radius * nx + posIntersect.x;
```

File: src/code/z_bgcheck.c [hunk 4]
```diff
@@ -1994,21 +1963,22 @@ s32 BgCheck_CheckWallImpl(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResul
...
-                n2XZDist = sqrtf(SQ(nx2) + SQ(nz2));
+                f32 nx2 = COLPOLY_GET_NORMAL(poly->normal.x);
+                f32 nz2 = COLPOLY_GET_NORMAL(poly->normal.z);
+                f32 n2XZDist = sqrtf(SQ(nx2) + SQ(nz2));
 
                 // if poly is not a "flat" floor or "flat" ceiling
                 if (!IS_ZERO(n2XZDist)) {
                     // normalize nx,nz and multiply each by the radius to go back to the other side of the wall
+                    f32 ny2;
                     f32temp = 1.0f / n2XZDist;
```
Comment (engineer124) [inline src/code/z_bgcheck.c:1974]:
```suggestion
                    f32 ny2;

                    f32temp = 1.0f / n2XZDist;
```

File: src/code/z_bgcheck.c [hunk 5]
```diff
@@ -2042,19 +2012,20 @@ s32 BgCheck_CheckWallImpl(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResul
     }
     // if a collision with a dyna poly was detected
     if (dynaPolyCollision || (*outBgId != BGCHECK_SCENE)) {
+        Vec3f posIntersect2;
+        s32 bgId2;
         if (BgCheck_CheckLineImpl(colCtx, xpFlags, COLPOLY_IGNORE_NONE, posPrev, posResult, &posIntersect2, &poly,
```
Comment (engineer124) [inline src/code/z_bgcheck.c:2017]:
```suggestion
        s32 bgId2;

        if (BgCheck_CheckLineImpl(colCtx, xpFlags, COLPOLY_IGNORE_NONE, posPrev, posResult, &posIntersect2, &poly,
```

File: src/code/z_bgcheck.c [hunk 6]
```diff
@@ -3348,7 +3313,8 @@ f32 BgCheck_RaycastFloorDyna(DynaRaycast* dynaRaycast) {
             vtxList = dynaRaycast->dyna->bgActors[*dynaRaycast->bgId].colHeader->vtxList;
 
             for (i2 = 0; i2 < 3; i2++) {
-                Math_Vec3s_ToVec3f(&vtx, &vtxList[COLPOLY_VTX_INDEX(poly->vtxData[i2])]);
+                Vec3s* src = &vtxList[COLPOLY_VTX_INDEX(poly->vtxData[i2])];
+                Math_Vec3s_ToVec3f(&vtx, src);
```
Comment (engineer124) [inline src/code/z_bgcheck.c:3317]:
```suggestion
                Vec3s* src = &vtxList[COLPOLY_VTX_INDEX(poly->vtxData[i2])];

                Math_Vec3s_ToVec3f(&vtx, src);
```

**Review by hensldm — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Pull request merged**